attributes/default.rb

```diff
@@ -91,3 +91,8 @@
 default['hopslog']['default']['private_ips']                      = ['10.0.2.15']
 default['hopsworks']['domain_truststore_path']                    = "#{node['hopsworks']['domain_truststore_path']}"
 default['hopsworks']['domain_truststore']                         = "#{node['hopsworks']['domain_truststore']}"
+
+# Hive config
+default['hive2']['metastore']['port']                             = "9083"
```

Comment:
@SirOibaf @Limmen  shouldn't these be defined in hive-chef? That means spark-chef would depend on hive-chef which is not ideal. How about moving this PR's code in a new spark-hive recipe in spark-chef?

Reply:
I'd totally make `spark-chef` dependent on `hive-chef` at the end of the day spark is using hive to provide a service to the user. This for me can be classified as dependency. 
Moving the code to a different recipe won't solve the attribute duplication issue.

Reply:
But I guess the dependency needs to be established anyway, regardless of a new recipe or not. I'm not hard minded on the latter either.

Reply:
> My thinking is that it's not necessary to install hive to have spark, it's a bit of an overkill for the majority of installations that won't use it.
> I agree. However, this configuration does not mean it is necessary to install hive to have spark, it just means that the hive-config is placed in spark so that if the user tries to connect to Hive without specifying hive properties in the code it will default to this hive-config. Spark installed using this recipe will run fine even without Hive installed. I'll look into refactoring out this hive config to its own recipe.

Reply:
1. What I'm saying is to add a dependency entry in the `metadata.rb` file and on top of the `attributes.rb` add `include_attributes 'hive2'`. What this means is that all the hive attributes are available to be use in the spark recipes. The only requirement in this case is that chef should be able to find the hive cookbook in the cookbooks directory. This happens automatically as vendoring `hopsworks-chef` will vendor also `hive-chef`. 
2. The problem of not running the hive recipe is not in the attribute file, but in the part of the code where you get the `private_recipe_ip` of the metastore. This won't work as if you don't run the recipe, Karamel won't add the IP of the metastore/hs2 to the json which is then passed to chef to run the recipe. 
If you want to make optional the `hive-site.xml` configuration for sparksql, because you want to make optional the installation of hive, then you have to make optional also the installation of the `feature store` as it relies heavily on `hive`. 
We want to make the feature store a flagship feature for the platform, as such, hive needs to be installed as part of the default installation.

Reply:
Ok then we can agree that from this jira onwards spark will depend on hive and we can keep this pr as it is.